dashboard/app/models/user.rb

```diff
@@ -620,8 +620,13 @@ def self.from_omniauth(auth, params)
       user.provider = auth.provider
       user.uid = auth.uid
       user.name = name_from_omniauth auth.info.name
-      user.email = auth.info.email
       user.user_type = params['user_type'] || auth.info.user_type
+      # Store emails, except when using Clever
+      user.email = auth.info.email unless user.user_type == 'student' && auth.provider == 'clever'
+
+      if auth.provider == 'clever' && User.find_by_email_or_hashed_email(user.email)
+        user.email = user.email + '.cleveremailalreadytaken'
```

Comment:
Possible bad edge case, I'm not sure though.

If we're creating a new student account, line 625 is a no-op so user.email is `nil` or blank?
Then if `find_by_email_or_hashed_email` finds a user with a blank email (presumably possible?) then line 628 sets the email to `nil.cleveremailalreadytaken` or just `.cleveremailalreadytaken`.  We'd also probably end up with lots of users with that email, which might cause validation errors/conflicts later?

Reply:
I just did a check on prod:

irb(main):013:0> User.find_by_email_or_hashed_email('')
=> nil
irb(main):014:0> User.find_by_email_or_hashed_email(nil)
=> nil

Seems like we might be okay?

Reply:
Will we stay okay though? Assuming 627 evaluates false we don't set user.email at all, so then in the future do we end up finding a user with an empty email?